src/ui/components/Video/imperative/updateGraphics.ts

```diff
@@ -0,0 +1,156 @@
+import { ExecutionPoint, ScreenShot } from "@replayio/protocol";
+
+import { PaintsCache, findMostRecentPaint } from "protocol/PaintsCache";
+import { RepaintGraphicsCache } from "protocol/RepaintGraphicsCache";
+import { pauseIdCache } from "replay-next/src/suspense/PauseCache";
+import { screenshotCache } from "replay-next/src/suspense/ScreenshotCache";
+import { ReplayClientInterface } from "shared/client/types";
+import { updateState } from "ui/components/Video/imperative/MutableGraphicsState";
+
+export async function updateGraphics({
+  abortSignal,
+  containerElement,
+  executionPoint,
+  replayClient,
+  time,
+}: {
+  abortSignal: AbortSignal;
+  containerElement: HTMLElement;
+  executionPoint: ExecutionPoint | null;
+  time: number;
+  replayClient: ReplayClientInterface;
+}) {
+  updateState(containerElement, {
+    executionPoint,
+    status: "loading",
+    time,
+  });
+
+  await PaintsCache.readAsync();
+
```

Comment:
I think we should check `abortSignal` here as well.

Reply:
Ah, yes. You're right.